.github/workflows/native-wsl.yml

```diff
@@ -0,0 +1,175 @@
+name: Native and WSL
+
+on: [push, pull_request]
+
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    defaults:
+        run:
+          shell: ${{ matrix.configuration == 'wsl' && 'wsl-bash {0}' || 'pwsh' }}
+    strategy:
+      matrix:
+        os: [windows-latest]
+        node-version: [16, 14, 12, 10]
+        configuration: [wsl, native]
+        architecture: [x86, x64]
+    steps:
+    - uses: actions/checkout@v4
+    - uses: Vampire/setup-wsl@v3
+      if: matrix.configuration == 'wsl'
+      with:
+        distribution: Ubuntu-22.04
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v4
+      if: matrix.configuration == 'native'
+      with:
+        node-version: ${{ matrix.node-version }}
+    - name: Do everything here because npm is not found when running in separate steps
+      if: matrix.configuration == 'wsl'
+      run: |
+        export ESLINT_VERSION=7
+        sudo apt update -y
+        sudo apt-get install curl -y
+        curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/master/install.sh | bash
+        export NVM_DIR="$HOME/.nvm"
+        [ -s "$NVM_DIR/nvm.sh" ] && \. "$NVM_DIR/nvm.sh"  # This loads nvm
+        [ -s "$NVM_DIR/bash_completion" ] && \. "$NVM_DIR/bash_completion"  # This loads nvm bash_completion
+        nvm install ${{ matrix.node-version }}
```

Comment:
if we're going to use nvm anyways, can we use the ljharb/actions/node/install action instead for WSL?

Reply:
Yes. But this is more transparent, which makes the whole easier to understand.

Reply:
It won't work because nvm won't be available across runs. That's why the whole script is in a single `run` instead of broken out into multiple runs. It looks like a limitation in the Vampire/setup-wsl@v3 action.

Reply:
are you sure? i use it across runs with that action on hundreds of repos.

Reply:
Yes. I was 100% sure. But now I'm questioning all of my life choices. Will you please send me some examples I can compare to?

Reply:
I tried using the `ljharb/actions/node/install@main` action again. Here's the [pipeline results](https://github.com/joeyguerra/eslint-plugin-import/actions/runs/8360381854/job/22885920314).

I think there's several parameters involved that are not working together correctly.

- The runner is a Windows runner. I think the `ljharb/actions/node/install@main` action is installing a node version for Windows, not Ubuntu
- The `Vampire/setup-wsl@v3` action is setting up WSL and installing Ubuntu
- Every `run` runs a new PowerShell (or cmd) "terminal" session and `Vampire/setup-wsl@v3` wraps the `run` scripts in a call to `wsl.BAT`, so previous `runs` that change "state" are not getting persisted for the next "session"

I'll tinker a bit more, but it's zapping my energy and motivation because I keep hitting different errors.

Reply:
ahh, interesting, ok. certainly my personal action doesn't usually run on windows so it might not work properly.